### ISSUE TITLE
visualize and unsupervised loss

### ISSUE THREAD

**Issue by huangyuan2020** [DrSleep/DenseTorch#11]
Thank you very much for your work, I find your examples are supervised, I wonder if I can use some custom unsupervised loss function to train, or do you have any suggestions?
In addition, I found that the training process was encapsulated in dt.engine, so could Tensorboardx be used to visualize the intermediate results of the training process?

**Comment by DrSleep** [comment]
1. You can specify which loss per each task to use. If you implement your own loss function to train, supervised or unsupervised, you can specify it in this example [here](https://github.com/DrSleep/DenseTorch/blob/dev/examples/src/losses.py#L6). By default all the loss functions accept the tensor with predictions and the tensor with ground truth during the forward pass (you can see some examples [here](https://github.com/DrSleep/DenseTorch/blob/dev/densetorch/engine/losses.py)). For the unsupervised case, you can provide some dummy ground truth and then implement the logic however you want.
2. At the moment you cannot visualise intermediate results of the training (or validation) process. As you rightly pointed out, the training process is hidden within `dt.engine.train` and neither inputs nor predictions are being used outside it. There is some WIP that adds an option of visualisation callback, but it is not a priority right now
